testing/e2e/clients/aks.go

```diff
@@ -38,16 +40,29 @@ type aks struct {
 	options                             map[string]struct{}
 }
 
+type ServicePrincipalOptions struct {
+	ApplicationObjectID          string
+	ApplicationClientID          string
+	ServicePrincipalObjectID     string
+	ServicePrincipalCredPassword string
+}
+
+type McOptFields struct {
+	ClusterName             string
+	Ctx                     context.Context
+	ServicePrincipalOptions *ServicePrincipalOptions
```

Comment:
I'm not 100% sure sp should be a mcoption. Maybe it should just be another parameter of the NewAKS function. 

`func NewAks(ctx context.Context, subscriptionId, resourceGroup, name, location string, spOpts *ServicePrincipalOptions, mcOpts ...McOpt) (*aks, error) {`

This function definition is pretty rough. We define spOpts in the function directly and in the mcOpts? That's not clean. Choose one place to define and do it well.

Reply:
agreed. moved it out from mcoptions completely